### ISSUE TITLE
OR clause syntax error

### ISSUE THREAD

**Issue by jberd126** [craiggwilson/fluent-mongo#47]
Thanks for supporting the FluentMongo project, it's much appreciated. I ran into the following issue however.

When I have this query:

```
IQueryable<Item> query = database.GetCollection<Item>("items").AsQueryable()
    .Where(e => e.DateTimes.Any(dt =>
        (dt.Begin <= begin && dt.End > begin) ||
        (dt.Begin >= begin && dt.Begin < end)) &&
        mappings.Keys.Contains(e.MapId));
```

The generated native Mongo query is invalid:

```
{ "dateTimes" : { "$elemMatch" : { "$or" : { "begin" : { "$lte" : ISODate("2012-01-05T07:00:00Z"), "$gte" : ISODate("2012-01-05T07:00:00Z"), "$lt" : ISODate("2012-01-06T07:00:00Z") }, "end" : { "$gt" : ISODate("2012-01-06T07:00:00Z") } } } }, 
"mapId" : { "$in" : [10036, 45733, 72018] } }
```

Resulting in the error:

```
Mongo query appears to be invalid. Error:
QueryFailure flag was $and/$or/$nor must be a nonempty array (response was { "$err" : "$and/$or/$nor must be a nonempty array", "code" : 13086 }).
```

The native Mongo query is missing opening JSON array demarcations surrounding 'or' clause: 

... { "$or" : [ { "begin" : ...

**Comment by craiggwilson** [comment]
Yes, this is a known issue.  It is fairly complex to deal with and I'm working on a way to handle this correctly.

**Comment by craiggwilson** [comment]
This should be fixed in release 1.3.1.1.

**Comment by jberd126** [comment]
Verified working in 1.3.1.1. Thanks!
